modules/enabling_encapsulation.adoc

```diff
@@ -0,0 +1,100 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/optimizing-cpu-usage.adoc
+
+[id="enabling-encapsulation_{context}"]
+= Working with mount namespace encapsulation
+
+As an OpenShift system administrator, you want `systemd` to consume less resources so that you can run more workloads on my system. You want to enable and disable this setting, and verify whether encapsulation is in effect.
+
+== Enabling and disabling encapsulation
+
+.Prerequisites
+
+* You have access to the cluster as a user with the cluster-admin role.
+
+* You have installed the OpenShift CLI (oc).
+
+* You have created the `MachineConfig` object.
+
+.Procedure
+
+. Add the following YAML to the `MachineConfig.yaml` file to control whether encapsulation is enabled or disabled:
+
++
+[source,yaml]
+----
+  apiVersion: machineconfiguration.openshift.io/v1
+  kind: MachineConfig
+  metadata:
+    labels:
+      machineconfiguration.openshift.io/role: master
+    name: 99-kubens-master
+  spec:
+    config:
+      ignition:
+        version: 2.2.0
+      systemd:
+        units:
+        - enabled: true <1>
+          name: kubens.service
+
+  apiVersion: machineconfiguration.openshift.io/v1
+  kind: MachineConfig
+  metadata:
+    labels:
+      machineconfiguration.openshift.io/role: worker
+    name: 99-kubens-worker
+  spec:
+    config:
+      ignition:
+        version: 2.2.0
+      systemd:
+        units:
+        - enabled: true <2>
+          name: kubens.service
+
+----
+<1> If `enabled` is set to true, encapsulation is in effect for the primary node. Both <1> and <2> need to be set to true. If either are false, encapsulation will not be enabled. If you wish to disable encapsulation, set both <1> and <2> to `false`.
+<2> If `enabled` is set to true, encapsulation is in effect for the worker node.
+
+== Verifying if encapsulation is enabled or disabled
+
+To establish whether of not encapsulation is enabled, issue the following commands:
+
+[source,yaml]
+----
+$ sudo readlink /proc/1/ns/mnt
+$ sudo readlink /proc/$(pgrep -n kubelet)/ns/mnt
+$ sudo readlink /proc/$(pgrep -n crio)/ns/mnt
+----
+
+This set of commands will return the mount namespaces associated with the Container Runtime, Kubelet and `crio`.
```

Comment:
The first command does not return the mount namespace of the Container Runtime; it returns the mount namespace of systemd.

The Container Runtime in OpenShift is crio.
```suggestion
This set of commands will return the mount namespaces associated with systemd, Kubelet and `crio`.
```

Reply:
Done.